IS0402Test.py

```diff
@@ -827,7 +845,11 @@ def test_31(self):
                                                                         .format(self.query_url,
                                                                                 curr_resource,
                                                                                 self.test_data[curr_resource]["id"]))
-                                if not v or r_resource_deleted.status_code != 404:
+                                if not v:
+                                    return test.FAIL(
+                                        "Query API returned an unexpected response: {}. Cannot execute test.".format(
+                                            r))
```

Comment:
r_resource_deleted?

Reply:
Well spotted. Hopefully the next commit sorts that case and a few others out